.github/workflows/ci.yaml

```diff
@@ -0,0 +1,176 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - "**"
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+**"
```

Comment:
I'd prefer a more strict pattern, ie. `1.0.0aoisdfaoishfoiahfoiadshfosidhfouis` shouldn't pass this match.

Reply:
The starting point here was that I want to support development tags as well as release tags (`0.1.2-dev.4` and `0.1.2` should both be supported).

There are significant limitations imposed on the [filter pattern](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) used here - for example grouping `()` and alternaitves `|` are not supported based on the documentation - so I tried to keep it simple.

This significantly hinders the ability to do repetition as required in the prerelease and build metadata parts.
I could either fix it on a single prerelease tag with `-[0-9a-zA-Z-]+` and `-[0-9a-zA-Z-]+.[0-9a-zA-Z-]+` and accept not running CI on multiple prerelease parts or do it for n pieces of tags or support everything (for which I opted originally).

Alternatively I could drop support for development tags entirely and use `[0-9]+.[0-9]+[0-9]+` which I'm hesitant to do, but I guess it can be argued people should only tag development tags from the default branch which should already have a passing CI, but then again, same could be said for tags, yet I see value in integration-testing tags (dev or release).

What solution do you propose in this case?

Reply:
We discussed externally to bind it to release and dev tags for now.

Fixed in d26c277.